complex/fabtest.h

```diff
@@ -180,25 +187,61 @@ enum ft_class_function {
 	FT_FUNC_COMPARE_ATOMIC,
 	FT_FUNC_COMPARE_ATOMICV,
 	FT_FUNC_COMPARE_ATOMICMSG,
-	FT_MAX_FUNCTIONS	
+	FT_MAX_FUNCTIONS
 };
 
 #define FT_FLAG_QUICKTEST	(1ULL << 0)
 
+#define is_inject_func(x) (x == FT_FUNC_INJECT || \
+			   x == FT_FUNC_INJECTDATA || \
+			   x == FT_FUNC_INJECT_WRITE || \
+			   x == FT_FUNC_INJECT_WRITEDATA || \
+			   x == FT_FUNC_INJECT_ATOMIC)
+
+#define is_read_func(x) (x == FT_FUNC_READ || \
+			 x == FT_FUNC_READV || \
+			 x == FT_FUNC_READMSG)
+
+#define is_fetch_func(x) (x == FT_FUNC_FETCH_ATOMIC || \
+			  x == FT_FUNC_FETCH_ATOMICV || \
+			  x == FT_FUNC_FETCH_ATOMICMSG)
+
+#define is_compare_func(x) (x == FT_FUNC_COMPARE_ATOMIC || \
+			    x == FT_FUNC_COMPARE_ATOMICV || \
+			    x == FT_FUNC_COMPARE_ATOMICMSG)
+
+#define is_data_func(x) (x == FT_FUNC_SENDDATA || \
+			 x == FT_FUNC_INJECTDATA || \
+			 x == FT_FUNC_WRITEDATA || \
+			 x == FT_FUNC_INJECT_WRITEDATA)
+
+#define is_msg_func(x)	(x == FT_FUNC_SENDMSG || \
+			 x == FT_FUNC_WRITEMSG || \
+			 x == FT_FUNC_READMSG || \
+			 x == FT_FUNC_ATOMICMSG || \
+			 x == FT_FUNC_FETCH_ATOMICMSG || \
+			 x == FT_FUNC_COMPARE_ATOMICMSG)
+
 struct ft_set {
 	char			node[FI_NAME_MAX];
 	char			service[FI_NAME_MAX];
 	char			prov_name[FI_NAME_MAX];
 	enum ft_test_type	test_type[FT_MAX_TEST];
 	enum ft_class_function	class_function[FT_MAX_FUNCTIONS];
+	uint64_t		msg_flags;
 	enum fi_op		op[FI_ATOMIC_OP_LAST];
+	enum fi_datatype	datatype[FI_DATATYPE_LAST];
 	enum fi_ep_type		ep_type[FT_MAX_EP_TYPES];
 	enum fi_av_type		av_type[FT_MAX_AV_TYPES];
 	enum ft_comp_type	comp_type[FT_MAX_COMP];
 	enum fi_wait_obj	eq_wait_obj[FT_MAX_WAIT_OBJ];
 	enum fi_wait_obj	cq_wait_obj[FT_MAX_WAIT_OBJ];
+	enum fi_resource_mgmt	resource_mgmt[FT_MAX_RES_MGMT];
+	enum fi_threading	threading[FT_MAX_THREADING];
```

Comment:
The test can iterate over threading values, but doesn't actually test threading.